worker-message-prioritization-target-capacity-calculators/src/main/java/com/github/workerframework/workermessageprioritization/targetqueue/K8sTargetQueueSettingsProvider.java

```diff
@@ -148,6 +155,14 @@ private TargetQueueSettings getTargetQueueSettingsFromKubernetes(final Queue tar
                                                    MESSAGE_PRIORITIZATION_TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_LABEL,
                                                    TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK));
                         targetQueueEligibleForRefillPercentage = TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK;
+                    } catch (final NumberFormatException ex) {
+                        // Eligible for refill percentage label is not a parsable long, set to fall back value
+                        LOGGER.error(String.format("Cannot get eligible for refill percentage for the %s queue. "
+                            + "An invalid %s label was provided. "
+                            + "Falling back to using max length of %s",
+                                                   targetQueueName, MESSAGE_PRIORITIZATION_TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_LABEL,
+                                                   TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK));
+                        targetQueueEligibleForRefillPercentage = TARGET_QUEUE_ELIGIBLE_FOR_REFILL_PERCENTAGE_FALLBACK;
```

Comment:
Could we use a common method to simplify this? Maybe something like:

```
    private static long getLabelOrDefault(
            final Map<String,String> labels,
            final String labelName,
            final String workerName,
            final String targetQueueName,
            final long defaultValue)
    {
        if (!labels.containsKey(labelName)) {
            LOGGER.error(
                    "Cannot get {} for the {} queue. The {} worker is missing the label. Falling back to using default value of {}",
                    labelName, targetQueueName, workerName, defaultValue);

            return defaultValue;
        }

        final String labelValue = labels.get(labelName);
        try {
            return Long.parseLong(labelValue);
        } catch (final NumberFormatException ex) {
            LOGGER.error(
                    "Cannot get {} for the {} queue. The {} worker provided an invalid (not parsable to long) label value: {}. " +
                            "Falling back to using default value of {}",
                    labelName, targetQueueName, workerName, labelValue, defaultValue);

            return defaultValue;
        }
    }
```

Reply:
Yes, this looks like a good change. Will implement and push up.